pkg/registry.go

```diff
@@ -3,27 +3,29 @@ package pkg
 import (
 	"fmt"
 	"math/big"
-	"strings"
 	"sync"
-
-	"github.com/ethereum/go-ethereum/accounts/abi"
 )
 
 // ProtocolRegistry maintains a registry of supported protocols and their operations.
 type ProtocolRegistry struct {
 	lock      sync.RWMutex
-	protocols map[ContractAddress]map[ContractAction]map[int64]ProtocolOperation
+	protocols map[ContractAddress]map[int64]ProtocolOperation
 }
 
 // NewProtocolRegistry creates a new instance of ProtocolRegistry.
-func NewProtocolRegistry() *ProtocolRegistry {
-	return &ProtocolRegistry{
-		protocols: make(map[ContractAddress]map[ContractAction]map[int64]ProtocolOperation),
+func NewProtocolRegistry(rpcURL string) *ProtocolRegistry {
+	r := &ProtocolRegistry{
+		protocols: make(map[ContractAddress]map[int64]ProtocolOperation),
 	}
+
+	r.setupProtocolOperations(rpcURL)
+	return r
 }
 
 // RegisterProtocolOperation registers a new operation for a protocol on a specific chain.
-func (pr *ProtocolRegistry) RegisterProtocolOperation(protocol ContractAddress, action ContractAction, chainID *big.Int, operation ProtocolOperation) {
+func (pr *ProtocolRegistry) RegisterProtocolOperation(protocol ContractAddress,
```

Comment:
Considering the more static approach, there could be reusability among the different versions of the same protocol. Consider, designing the API to track the `ProtocolName` and `Version` so that:
```Go
    aaveV4Op, err := registry.GetProtocolOperation("Aave", AaveV4)
    calldataV4, err := aaveV4Op.GenerateCalldata(...)
```

Reply:
This is an optional suggestion.

Reply:
https://github.com/blndgs/protocol_registry/pull/18/commits/1a4c73c39e9648d6936a0e56ff39f66e7d5ffc3a